code/drasil-gool/lib/Drasil/Shared/LanguageRenderer/LanguagePolymorphic.hs

```diff
@@ -206,6 +206,7 @@ objVar o' v' = do
         (variableType v) (R.objVar (RC.variable o) (RC.variable v))
   objVar' (variableBind v)
 
+-- This seems a bit sketchy - unwrapping everything to Doc, mixing it up, then wrapping it back to Variable
```

Comment:
I think this is supposed to create something that points to an array element, and then will be used as if it were a variable.

This is likely only used on the LHS, so that a redesign of L-Values might get rid of this.